### PR TITLE
Update all dependencies

### PR DESCRIPTION
**Pull request by camh-**
Update all Go dependencies with:

    go get -u all
    go mod tidy

This is mostly to get go-jsonnet v0.17.0 so I can get some new library
functions for my tests, but may as well keep up-to-date.

This has caused some new lint warnings as EvaluateSnippet has become
deprecated. Update to use the non-deprecated EvaluateAnonymousSnippet
instead. Also fix a silly linter error that can't read godocs properly
(I think it thought the comment line starting with `// package: ` was a
compiler/tool directive and not just part of the comment, so it ignored
that and the period that was at the end, and then complained about a
missing period. Changing the `:` to ` -` has removed that warning.

Update golangci-lint-action from v1 to v2.

Update the lint `Makefile` snippet to the latest that no longer uses Docker,
but builds the version locally instead.

Bump lint to v1.37.0 - the current latest version.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,22 +50,20 @@ FAIL_COVERAGE = { echo '$(COLOUR_RED)FAIL - Coverage below $(COVERAGE)%$(COLOUR_
...
+	$(GOLINT) run
 
-lint-with-docker:  ## Lint source code with docker image of golangci-lint
-	docker run --rm -w /src \
-		-v $(shell pwd):/src -v $(GOPATH1):/go -v $(HOME)/.cache:/root/.cache \
-		golangci/golangci-lint:v$(GOLINT_VERSION) \
-		golangci-lint run
+$(GOBIN)/$(GOLINT):
+	cd /tmp; \
+	GOBIN=/tmp GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v$(GOLINT_VERSION); \
```
Comment (juliaogris) [inline Makefile:63]: If this is an upgrade pr, maybe this could also mandate go 1.16 now and use go install without and cd tmp?

**Review by juliaogris — APPROVED** [review]
lgtm with one comment.

File: Makefile (hunk 1 above)
Comment (camh-) [reply Makefile]: Done. PTAL

File: Makefile [hunk 2]
```diff
@@ -50,22 +50,19 @@ FAIL_COVERAGE = { echo '$(COLOUR_RED)FAIL - Coverage below $(COVERAGE)%$(COLOUR_
...
+	$(GOLINT) run
 
-lint-with-docker:  ## Lint source code with docker image of golangci-lint
-	docker run --rm -w /src \
-		-v $(shell pwd):/src -v $(GOPATH1):/go -v $(HOME)/.cache:/root/.cache \
-		golangci/golangci-lint:v$(GOLINT_VERSION) \
-		golangci-lint run
+$(GOBIN)/$(GOLINT):
+	GOBIN=/tmp go install github.com/golangci/golangci-lint/cmd/golangci-lint@v$(GOLINT_VERSION) && \
+	      mv /tmp/golangci-lint $@
```
Comment (juliaogris) [inline Makefile:63]: nice!

**Review by juliaogris — APPROVED** [review]
and simpler again, great.
I wonder if you should change go version in go.mod to 1.16 as well?

**Comment by camh-** [comment]
> I wonder if you should change go version in go.mod to 1.16 as well?

I dont think so. Nothing in the packages require that. Just the makefile. I don't want to force anything that uses this package as an import to require go 1.16.

**Pull request merged**